bin/cli.js

```diff
@@ -1,116 +1,192 @@
+    console.log(`\n🔍 MCP Inspector is up and running at http://localhost:${CLIENT_PORT}${portParam} 🚀`);
+    try {
+        await Promise.any([server, client]);
+    }
+    catch (e) {
+        if (!cancelled || process.env.DEBUG) {
+            throw e;
+        }
+    }
+}
+async function runCli(args) {
+    const projectRoot = resolve(__dirname, "..");
+    const cliPath = resolve(projectRoot, "cli", "build", "index.js");
+    const abort = new AbortController();
+    let cancelled = false;
+    process.on("SIGINT", () => {
+        cancelled = true;
+        abort.abort();
+    });
+    try {
+        await spawnPromise("node", [cliPath, args.command, ...args.args], {
+            env: { ...process.env, ...args.envArgs },
+            signal: abort.signal,
+            echoOutput: true,
+        });
+    }
+    catch (e) {
+        if (!cancelled || process.env.DEBUG) {
+            throw e;
+        }
+    }
+}
+function loadConfigFile(configPath, serverName) {
+    try {
+        const resolvedConfigPath = path.isAbsolute(configPath)
+            ? configPath
+            : path.resolve(process.cwd(), configPath);
+        if (!fs.existsSync(resolvedConfigPath)) {
+            throw new Error(`Config file not found: ${resolvedConfigPath}`);
+        }
+        const configContent = fs.readFileSync(resolvedConfigPath, "utf8");
+        const parsedConfig = JSON.parse(configContent);
+        if (!parsedConfig.mcpServers || !parsedConfig.mcpServers[serverName]) {
+            const availableServers = Object.keys(parsedConfig.mcpServers || {}).join(", ");
+            throw new Error(`Server '${serverName}' not found in config file. Available servers: ${availableServers}`);
+        }
+        const serverConfig = parsedConfig.mcpServers[serverName];
+        return serverConfig;
+    }
+    catch (err) {
+        if (err instanceof SyntaxError) {
+            throw new Error(`Invalid JSON in config file: ${err.message}`);
+        }
+        throw err;
+    }
+}
+function parseKeyValuePair(value, previous = {}) {
+    const parts = value.split("=");
+    const key = parts[0];
+    const val = parts.slice(1).join("=");
+    if (val === undefined || val === "") {
+        throw new Error(`Invalid parameter format: ${value}. Use key=value format.`);
+    }
+    return { ...previous, [key]: val };
+}
+function parseArgs() {
+    const program = new Command();
+    const argSeparatorIndex = process.argv.indexOf("--");
+    let preArgs = process.argv;
+    let postArgs = [];
+    if (argSeparatorIndex !== -1) {
+        preArgs = process.argv.slice(0, argSeparatorIndex);
+        postArgs = process.argv.slice(argSeparatorIndex + 1);
+    }
+    program
+        .name("inspector-bin")
+        .allowExcessArguments()
+        .allowUnknownOption()
+        .option("-e <env>", "environment variables in KEY=VALUE format", parseKeyValuePair, {})
+        .option("--config <path>", "config file path")
```

Comment:
This config file here is the standard `mcpServers` json file right ? Are you planning to load any other config from this file ? If not should this argument be called `--mcp-server` or something similar ?

And it can take the `file:<server_name` as a single argument (instead of a separate `--server` arg which is always required)

So:
`npx @modelcontextprotocol/inspector --mcp-server "path/to/config.json:everything"`